### PR TITLE
Reworked scroll definitions.

### PR DESCRIPTION
**Pull request by Jaccouille**
$changelog: Changed Scroll of Swiftness to Scroll of Haste, which now affect nearby allies as well.
$changelog: Changed Scroll of Entangling Root to only ensnare trolls.

I didn't change tsunami extra damage code, I believe it was done for balance purpose.

### CODE REVIEW

File: wurst/objects/items/ScrollsDefinition.wurst [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+// Create 2 skeleton warrior following the caster
+// Skeleton warrior got the locust ability, meaning they can't be targetted/attacked and randomly move around, attacking surrounding units
+@compiletime function createSkeletalMinionAbility() returns AbilityDefinitionCryptLordLocustSwarm
+    return new AbilityDefinitionCryptLordLocustSwarm(ABILITY_SKELETAL_MINION)
+        ..setName("Scroll of Living Dead")
+        ..setCheckDependencies(false)
+        ..setDamageReturnFactor(1, 0)
+        ..setNumberofSwarmUnits(1, 2)
+        ..setDamageReturnThreshold(1, 600)
+        ..setMaxSwarmUnitsPerTarget(1, 5)
```
Comment (jonathanmclaus) [inline wurst/objects/items/ScrollsDefinition.wurst]: Is this maximum tracked across multiple scrolls? If three trolls each use a scroll next to the same enemy, can that enemy be focused by all 6 skeletons?

File: wurst/objects/items/ScrollsDefinition.wurst [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+        ..setLumberCost(9)
+        ..setAbilities(ABILITY_CYCLONE.toRawCode())
+        ..setStockReplenishInterval(120)
+        ..setModelUsed(LocalItems.scrollCyan)
+        ..setInterfaceIcon(Icons.bTNBansheeMaster)
+        ..setNameEnhance("Scroll of Cyclone")
+        ..setTooltipExtended(SC_TT)
+
+
+let BUFF_CYCLONE = 'Bcyc'
```
Comment (jonathanmclaus) [inline wurst/objects/items/ScrollsDefinition.wurst]: Can you leave a comment explaining why both of these are needed?

File: wurst/objects/items/ScrollsDefinition.wurst [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+        if not target.hasAbility(BUFF_CYCLONE) and not target.hasAbility(BUFF_CYCLONE_EXTRA)
+            target.removeAbility(GHOST_VIS_ID)
+            buffPoller.stop()
+
+function unit.isFire() returns bool
+    return this.getTypeId() == UNIT_FIRE
+        or this.getTypeId() == UNIT_MAGE_FIRE_SUMMONED
+        or this.getTypeId() == UNIT_MAGE_FIRE
+
+function onTsunami(unit caster, vec2 target)
```
Comment (jonathanmclaus) [inline wurst/objects/items/ScrollsDefinition.wurst]: I thought you were using 2 cone abilities tied to a single dummy spell for this. Did you decide not to?

File: wurst/objects/items/ScrollsDefinition.wurst [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+                u.kill()
+            // Otherwise deal additional damage.
+            else
+                caster.damageTarget(u, TSUNAMI_STRUCT_DMG)
+
+function onRoot(unit caster, unit target)
+    if target.isTroll()
+        InstantDummyCaster.castTarget(caster.getOwner(), ABILITY_ROOT_TROLL, 1, Orders.ensnare, target)
+    else
+        InstantDummyCaster.castTarget(caster.getOwner(), ABILITY_ROOT_HOSTILE, 1, Orders.entanglingroots, target)
```
Comment (jonathanmclaus) [inline wurst/objects/items/ScrollsDefinition.wurst]: Any ability that does damage should not be cast using an instant dummy caster. The instant dummy caster can change owner immediately after it casts the spell, meaning the damage and the kill are made for the wrong player.

File: wurst/objects/items/ScrollsDefinition.wurst (hunk 1 above)
Comment (Jaccouille) [reply wurst/objects/items/ScrollsDefinition.wurst]: > Is this maximum tracked across multiple scrolls?
Yes, enemy can be focused by 5 skeleton max.

File: wurst/objects/items/ScrollsDefinition.wurst (hunk 3 above)
Comment (Jaccouille) [reply wurst/objects/items/ScrollsDefinition.wurst]: I decided not to, didn't work at my first try so I chose the lazy path and let it this way.

File: wurst/objects/items/ScrollsDefinition.wurst (hunk 2 above)
Comment (Jaccouille) [reply wurst/objects/items/ScrollsDefinition.wurst]: Updated.

File: wurst/objects/items/ScrollsDefinition.wurst (hunk 4 above)
Comment (Jaccouille) [reply wurst/objects/items/ScrollsDefinition.wurst]: Updated.

File: wurst/objects/items/ScrollsDefinition.wurst [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+                u.kill()
+            // Otherwise deal additional damage.
+            else
+                caster.damageTarget(u, TSUNAMI_STRUCT_DMG)
+
+function onRoot(unit caster, unit target)
+    if target.isTroll()
+        InstantDummyCaster.castTarget(caster.getOwner(), ABILITY_ROOT_TROLL, 1, Orders.ensnare, target)
+    else
+        new DummyCaster()
```
Comment (jonathanmclaus) [inline wurst/objects/items/ScrollsDefinition.wurst]: Don't you need to set its lifespan?

Comment (Jaccouille) [reply wurst/objects/items/ScrollsDefinition.wurst]: It's 5 by default, but I guess I should fix it to the root duration.

File: wurst/objects/items/ScrollsDefinition.wurst [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+            ..owner(caster.getOwner())
+            ..delay(ROOT_DURATION_NORMAL)
+            ..origin(caster.getPos())
+            ..castTarget(ABILITY_ROOT_HOSTILE, 1, Orders.entanglingroots, target)
+
+
+init
+    EventListener.onTargetCast(ABILITY_CYCLONE, (unit caster, unit target) -> onCyclone(target))
+    EventListener.onPointCast(ABILITY_TSUNAMI, (unit caster, vec2 target) -> onTsunami(caster, target))
```
Comment (jonathanmclaus) [inline wurst/objects/items/ScrollsDefinition.wurst]: I don't think you need to create a trivial lambda here. Does the following work?

    EventListener.onTargetCast(ABILITY_TSUNAMI, function onTsunami)

File: wurst/systems/crafting/QuickMakeToolTips.wurst [hunk 7]
```diff
@@ -177,7 +177,7 @@ public let SCROLL_FIREBALL_RECIPE       = "a Fireball Spell Scroll. Can cast fir
 public let SCROLL_CYCLONE_RECIPE        = "a Cyclone Spell Scroll. Can cast cyclone on a target enemy."+makeToolTipDuration(5, 10, 30)+"\n\n1x "+WIND+" + 1x "+MC
 public let SCROLL_LIVING_DEAD_RECIPE    = "a Living Dead Spell Scroll. Can summon 2 skeletons fighting for you."+makeToolTipDuration(30, 30, 60)+"\n\n1x "+BONE+" + 1x "+MC
 public let SCROLL_TSUNAMI_RECIPE        = "a Tsunami Spell Scroll. Can cast a tsunami, dealing 25 damage. Good for destroying base defenses."+makeToolTipCooldown(25)+"\n\n1x "+WATER+" + 1x "+MC
-public let SCROLL_SWIFTNESS_RECIPE      = "a Swiftness Spell Scroll. Can increase movement speed of surrounding allies to maximum."+makeToolTipDuration(7, 7, 35)+"\n\n1x "+ANABOLIC+" + 1x "+MC
+public let SCROLL_HASTE_RECIPE      = "a HASTE Spell Scroll. Can increase movement speed of surrounding allies to maximum."+makeToolTipDuration(7, 7, 35)+"\n\n1x "+ANABOLIC+" + 1x "+MC
```
Comment (jonathanmclaus) [inline wurst/systems/crafting/QuickMakeToolTips.wurst]: This didn't play nicely with your find + replace. The operator isn't lined up and `HASTE` is capitalized.

**Pull request merged**